src/BlazorWasmSdk/Tasks/ComputeBlazorPublishAssets.cs

```diff
@@ -193,6 +194,22 @@ public override bool Execute()
                     {
                         newDotNetJs = new TaskItem(Path.GetFullPath(aotDotNetJs.ItemSpec), asset.CloneCustomMetadata());
                         newDotNetJs.SetMetadata("OriginalItemSpec", aotDotNetJs.ItemSpec);
+
+                        var existingRelativePath = newDotNetJs.GetMetadata("RelativePath");
+                        var existingRelativePathWithoutExtension = Path.Combine(Path.GetDirectoryName(existingRelativePath), Path.GetFileNameWithoutExtension(existingRelativePath));
+
+                        // A Base64 encoded hash can contain `/`, `=` and `+` which may be problematic for URLs/file names.
+                        // As we're only using the hash for the purposes of cache busting, we can opt to simply remove those chars 
+                        // from the hash (at the cost of a tiny bit of entropy).
+                        //
+                        // URL encoding is not an option as encoded file names are decoded at another layer and not
+                        // reflected in the final published asset file name (ie. final asset would still contain symbols).
+                        //
+                        // https://en.wikipedia.org/wiki/Base64#Base64_table
+                        var itemHash = GetItemHash(aotDotNetJs).Replace("/", string.Empty).Replace("=", string.Empty).Replace("+", string.Empty);
+                        var newRelativePath = $"{existingRelativePathWithoutExtension}.{itemHash}.js";
```

Comment:
is picking a substring (first 8 / 10 chars) sufficient? We don't need to tack on 40+ chars to the string to make it unique.

Reply:
Agree